File: maximum-and-minimum-in-an-array.cpp
//Time Complexity - O(n)
//Space complexity - O(1)

class Solutions {
	pair<int,int> minMax(vector<int> nums){
		int Min , Max;

		if(nums[0]>nums[1]){
			Min = nums[1];
			Max = nums[0];
		} else{
			Min = nums[0];
			Max = nums[1];
		}

		for(int i=2;i<nums.size()-1;i+2){
			if(nums[i]>nums[i+1]){
				Min = min(Min,nums[i+1]);
				Max = max(Max,nums[i]);
			} else {
				Min = min(Min,nums[i]);
				Max = max(Max,nums[i+1]);
			}
		}

		if(nums.size()%2==1){
			Min = min(Min,nums[nums.size()-1]);
			Max = max(Max,nums[nums.size()-1]);
		}

		return make_pair(Min,Max);
	}
}

File: maximum-and-minimum-in-an-array.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : No, cannot find in problem in leetcode
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
// if update max if element id greater than max. If element is greater than max then we do not have to check for min.
// else check for min.


public void MaxAndMinInArray(int[] arr){
    if(arr == null || arr.length == 0) return;
    int min = arr[0];
    int max = arr[0];
    for(int  i = 0; i < arr.length; i++){
        if(arr[i]>max) max = arr[i];
        else if(arr[i]<min) min = arr[i];
    }
    System.out.println("Min : " + min + " Max : " + max);
}